resolver/src/dns_lru.rs

```diff
@@ -35,11 +35,38 @@ impl LruValue {
 }
 
 #[derive(Debug)]
-pub(crate) struct DnsLru(LruCache<Query, LruValue>);
+pub(crate) struct DnsLru {
+    cache: LruCache<Query, LruValue>,
+    /// An optional minimum TTL value for positive responses.
+    ///
+    /// Positive responses with TTLs under `min_positive_ttl` will use
+    /// `` instead.
+    min_positive_ttl: Duration,
+    /// An optional minimum TTL value for negative (`NXDOMAIN`) responses.
+    ///
+    /// `NXDOMAIN` responses with TTLs under `min_negative_ttl` will use
+    /// `min_negative_ttl` instead.
+    min_negative_ttl: Duration,
+}
 
 impl DnsLru {
     pub(crate) fn new(capacity: usize) -> Self {
-        DnsLru(LruCache::new(capacity))
+        let cache = LruCache::new(capacity);
+        DnsLru {
+            cache,
+            min_positive_ttl: Duration::from_secs(0),
+            min_negative_ttl: Duration::from_secs(0),
+        }
+    }
+
+    pub(crate) fn with_min_positive_ttl(mut self, min: Duration) -> Self {
+        self.min_positive_ttl = min;
+        self
+    }
+
+    pub(crate) fn with_min_negative_ttl(mut self, min: Duration) -> Self {
+        self.min_negative_ttl = min;
+        self
```

Comment:
NIT: It's better to set the minimums once and then not let them be changed. Then you avoid having to answer questions like "what should we do with all the existing items in the cache when the new minimums are less than the old minimums?"

Reply:
I agree, but I also didn't like having a constructor that takes two `Option<Duration>`s as positional arguments, since then the caller is required to remember the ordering and whether the first or second argument corresponds to positive or negative results. I went back and forth on it for a while & decided having functions which name explictly which TTL is being set was better. 

Probably the Right Thing To Do here would be to have a builder, so that we get both the explict knowledge of which minimum TTL is being set _and_ the enforcement that they're set once and never changed...

Reply:
You could also have `struct PositiveMinTTL<T>(T)` and `struct NegativeMinTTL<T>(T)` so that the caller would do `DnsLru::new(whatever, PositiveMinTTL(min), NegativeMinTTL(max))`. I've seen a number of libraries follow that pattern before and it avoids needing to create a builder. (I personally do not like the builder pattern because it turns a stateless thing into a stateful thing.)

Reply:
Okay, PR fef5f93c11a6e46e2e4e5aae5245ea00e0e3de2b changes this so the minimum TTLs are configured using a builder. This way, they can't be changed after the cache is constructed, but which TTL is being configured is less ambiguous than it would be with positional arguments.

This should also make our lives _much_ easier when we're implementing #489; so we don't end up with a constructor taking `(usize, Option<Duration>, Option<Duration>, Option<Duration>, Option<Duration>)` :P

Reply:
Actually, while editing that comment, I turned it into something kind of senseless. You could just have `struct TtlLimits { min_positive: Option<Duration>, min_negative: Option<Duration> }`. That would be obvious to extend with maximums later. (This is an internal API IIRC so it doesn't have to be perfect, regardless.)

Reply:
Personally, for the builder patter I always do a separate type for the builder. I also like the type safe TTL...

Reply:
I don't want to derail the review  with any suggestions. I am fine either way with how it's implemented.

Reply:
I'm going to take @briansmith's suggestion of a configuration struct rather than the builder; sorry for the amount of churn on this PR...

Reply:
Changed the constructor parameter to a struct in bf74a6e4fb417c2d232d289aaf7905d36a6631d5.